### ISSUE TITLE
Batch execution

### ISSUE THREAD

**Issue by Keats** [blackbeam/rust-mysql-simple#37]
Hey I'm working on a SQL migration CLI (https://github.com/Keats/dbmigrate) and just started adding mysql support.

Is there a method to run several statements in one go?
Such as:
```rust
        self.pool.prep_exec("
            CREATE TABLE IF NOT EXISTS migrations_table(id INTEGER, current INTEGER);
            INSERT INTO migrations_table (id, current)
            SELECT 1, 0 FROM DUAL
            WHERE NOT EXISTS(SELECT * FROM migrations_table WHERE id = 1);
        ", ()).unwrap();
```
which currently fails with:
```
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'INSERT INTO migrations_table (id, current)
            SELECT 1, 0 FROM DUAL
   ' at line 2', ../src/libcore/result.rs:738

```

(and obviously the SQL migration files would contain several statements)

The postgres crate provides `batch_execute` for that purpose

**Comment by blackbeam** [comment]
`prep_exec` is a shortcut for `prepare` and `execute`.
AFAIK mysql does not allow to prepare statements with more than one SQL statement in it.

You still can execute multiple SQL statements at once using mysql's text protocol (with [`MyPooledConn::query`](http://blackbeam.org/doc/mysql/conn/pool/struct.MyPooledConn.html#method.query) method).

```rust
self.pool.get_conn().and_then(|conn| {
    conn.query(
        r"CREATE TABLE IF NOT EXISTS migrations_table(id INTEGER, current INTEGER);
        INSERT INTO migrations_table (id, current)
        SELECT 1, 0 FROM DUAL
        WHERE NOT EXISTS(SELECT * FROM migrations_table WHERE id = 1);"
    ).map(|_| ())
});
```

**Comment by Keats** [comment]
Cheers, that worked.
Release a new version with mysql support

**Comment by ufoscout** [comment]
@blackbeam 
I used the approach you proposed for batch inserts; however, after that, it seems that if I reuse the connection for whatever query I always get a DriverError::PacketOutOfSync. Do you have any clue on it?

**Comment by blackbeam** [comment]
Hi, @ufoscout.

`DriverError::PacketOutOfSync` means that protocol is messed up and this might be a bug in the library.
Could you please provide more details?

**Comment by ufoscout** [comment]
@blackbeam 
I created a reproducer for the issue here:
https://github.com/ufoscout/mysql_rust_issue

launch: `cargo test` and it fails with:

```
---- test::batch_execute stdout ----
thread 'test::batch_execute' panicked at 'called `Result::unwrap()` on an `Err` value: DriverError { Packet out of sync }', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

The test starts a docker container with MySql 5.7.25 and executes the queries on it; so, you need to have docker installed locally. 
If not present locally, it will download the mysql docker images so, the first time, it could take some time. After that, the docker image will start in more or less 5 seconds.
I tested it on:
- Ubuntu 18.04 LTS, 64 bit
- rust 1.34

**Comment by blackbeam** [comment]
@ufoscout, thanks for your work!

This issue is fixed in [v16.0.1](https://github.com/blackbeam/rust-mysql-simple/releases/tag/v16.0.1) and [v15.1.1](https://github.com/blackbeam/rust-mysql-simple/releases/tag/v15.1.1).

Your test in https://github.com/ufoscout/mysql_rust_issue should work after `cargo update`.

**Comment by ufoscout** [comment]
@blackbeam 
great! I confirm it works
